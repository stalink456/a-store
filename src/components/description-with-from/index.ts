export { DescriptionWithForm } from './description-with-form';
