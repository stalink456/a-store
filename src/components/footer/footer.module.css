.footer {
  padding: var(--gap-xl) 0;
  margin: 0 auto;
}
