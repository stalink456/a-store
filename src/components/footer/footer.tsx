import React from "react";
import { Space } from "@alfalab/core-components/space";
import { Typography } from "@alfalab/core-components/typography";

import styles from "./footer.module.css";

export const Footer: React.FC = () => {
  return (
    <div className={styles.footer}>
      <Space direction="horizontal" fullWidth>
        <Typography.Text tag="p" weight="medium" color="disabled">
          © ООО «Альфа Фьюче Пипл», 2023
        </Typography.Text>
      </Space>
    </div>
  );
};
