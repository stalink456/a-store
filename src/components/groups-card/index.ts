export { GroupsCard } from './groups-card';
