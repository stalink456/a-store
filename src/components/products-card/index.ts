export { ProductsCard } from "./products-card";
