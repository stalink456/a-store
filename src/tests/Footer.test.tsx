import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Footer } from 'components/footer';

describe('Footer component', () => {
  it('should render Footer component', () => {
    const { baseElement } = render(<Footer />, { wrapper: MemoryRouter });
    expect(baseElement).toMatchSnapshot();
  });

  it("shouldn't render Footer component with non-existent text", () => {
    render(<Footer />, { wrapper: MemoryRouter });

    const result = screen.queryByText('non-existent footer');
    expect(result).not.toBeInTheDocument();
  });
});
