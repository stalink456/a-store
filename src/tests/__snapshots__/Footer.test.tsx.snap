// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render Footer component 1`] = `
<body>
  <div>
    <div
      class="footer"
    >
      <div
        class="space__spaceContainer_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as"
      >
        <div
          class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
        >
          <p
            class="typography__paragraph_c16y5 typography__disabled_hpb71 typography__primary-medium_c16y5 typography__medium_c16y5"
          >
            © ООО «Альфа Фьюче Пипл», 2023
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
